substrate/frame/session/src/lib.rs

```diff
@@ -607,33 +603,53 @@ pub mod pallet {
 }
 
 impl<T: Config> Pallet<T> {
+	/// Public function to access the current set of validators.
+	pub fn validators() -> Vec<T::ValidatorId> {
```

Comment:
We can already mark these as deprecated, and our codebase will be fine right?

Reply:
You mean not using the `validators` function?

It is being used in a lot of places outside of the session pallet like [this](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/babe/src/tests.rs#L435)

Reply:
I would say that deprecating the compatibility functions and changing them within the sdk is in scope for this PR. I had a quick flick through a couple of PRs from the linked issue and others have replaced usage of getters outside the pallet in similar situations.

Reply:
It's a little dependent. Even I had two ways on this. Some PRs have changed, some haven't. If this sounds, good I can replace all instances (outside the pallet as well). 
cc: @kianenigma